File: framework/languages/node/packages/framework/src/contracts/Spots/SpotKind.ts
export enum ZLinkSpotKind {
  Invalid = 'invalid',
  Entry = 'entry',
  User = 'user',
  Instance = 'instance'
}

export function zlinkSpotKindToWire(kind: ZLinkSpotKind): number {
  switch (kind) {
    case ZLinkSpotKind.Entry: return 1;
    case ZLinkSpotKind.User: return 2;
    case ZLinkSpotKind.Instance: return 3;
    default: return 0;
  }
}

export function zlinkSpotKindFromWire(value: number): ZLinkSpotKind {
  switch (value) {
    case 1: return ZLinkSpotKind.Entry;
    case 2: return ZLinkSpotKind.User;
    case 3: return ZLinkSpotKind.Instance;
    default: return ZLinkSpotKind.Invalid;
  }
}

File: framework/languages/java/zlink-framework-binding-internal/src/main/java/systems/zlink/framework/runtime/internal/binding/spot/ServiceEnums/SpotKind.java
/* SPDX-License-Identifier: MPL-2.0 */

package systems.zlink.framework.runtime.internal.binding.spot;

/** The kind of a spot. */
enum SpotKind {
    /** No spot kind (unset). */
    INVALID,
    /** A node's well-known entry spot. */
    ENTRY,
    /** A user-created spot. */
    USER
}
